### PR TITLE
[HDInsight on AKS] New Api version 2023-11-01-preview

### PR DESCRIPTION
**Pull request by yuc-Li**
## Upcoming Release
- Changes in the Cluster Pool
    - Enable create cluster pool with user network profile.
    - Enable get cluster pool available upgrade versions.
    - Enable upgrade cluster pool.
- Changes in the Cluster
    - Enable create Ranger cluster.
    - Enable get cluster available upgrade versions.
    - Enable set internal ingress.
    - Enable upgrade cluster.
- Added commands
  - Get-AzHdInsightOnAksClusterAvailableUpgrade
  - Get-AzHdInsightOnAksClusterPoolAvailableUpgrade
  - Update-AzHdInsightOnAksClusterPool
  - New-AzHdInsightOnAksClusterHotfixUpgradeObject
  - New-AzHdInsightOnAksClusterPoolAKSUpgradeObject 

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by yuc-Li** [comment]
Hi @vidai-msft, I use autorest.powershell to generate the latest code has syntax errors, I cannot successfully build and continue development, could you help take a look at it.
image

**Comment by vidai-msft** [comment]
@yuc-Li Please do NOT update the version number in the PR.
For the codegen error, you can try following commands.
* autorest --reset
* autorest

**Comment by yuc-Li** [comment]
> @yuc-Li Please do NOT update the version number in the PR. For the codegen error, you can try following commands.
> 
> * autorest --reset
> * autorest

I have modified the change of the version number and tried to build again after reset autorest. Still have same issues.

**Comment by yuc-Li** [comment]
> > @yuc-Li Please do NOT update the version number in the PR. For the codegen error, you can try following commands.
> > 
> > * autorest --reset
> > * autorest
> 
> I have modified the change of the version number and tried to build again after reset autorest. Still have same issues.

image

**Comment by yuc-Li** [comment]
@vidai-msft , could you help on this? We want to participate in the 4/2nd release. Thanks.

**Comment by yuc-Li** [comment]
Hi @vidai-msft , could you help take a look on this build error. These values were not found in Branch generation.
```
D:\a\_work\1\s\src\Accounts\Authentication\Authentication\Parameters\AuthenticationParameters.cs(53,87): error CS0117: 'ConfigKeys' does not contain a definition for 'DisableInstanceDiscovery' [D:\a\_work\1\s\src\Accounts\Authentication\Authentication.csproj] [D:\a\_work\1\s\build.proj]
D:\a\_work\1\s\src\Accounts\Authentication\Config\Definitions\DisableInstanceDiscoveryConfig.cs(32,50): error CS0117: 'ConfigKeys' does not contain a definition for 'DisableInstanceDiscovery' [D:\a\_work\1\s\src\Accounts\Authentication\Authentication.csproj] [D:\a\_work\1\s\build.proj]
    Az.HdInsightOnAks -> D:\a\_work\1\s\artifacts\Debug\Az.HdInsightOnAks\HdInsightOnAks.Autorest\bin\Az.HdInsightOnAks.private.dll
  
  Build FAILED.
```

File: src/HdInsightOnAks/HdInsightOnAks.Autorest/README.md [hunk 1]
```diff
@@ -49,14 +46,15 @@ In this directory, run AutoRest:
...
+tag: package-preview-2023-11
+commit: 754d31188c080f20b8155acdf8b8fbd77dace467
 require:
   - $(this-folder)/../../readme.azure.noprofile.md
   - $(repo)/specification/hdinsight/resource-manager/Microsoft.HDInsight/HDInsightOnAks/readme.md
 
 # input-file:
-#  - D:\source\azure-rest-api-specs\specification\hdinsight\resource-manager\Microsoft.HDInsight\HDInsightOnAks\preview\2023-06-01-preview\hdinsight.json
-inlining-threshold: 100
+#  - D:\HDI-Repo\azure-rest-api-specs\specification\hdinsight\resource-manager\Microsoft.HDInsight\HDInsightOnAks\preview\2023-11-01-preview\hdinsight.json
```
Comment (vidai-msft) [inline src/HdInsightOnAks/HdInsightOnAks.Autorest/README.md:56]: Local input file is not qualified. Please replace with a link from the swagger repo.

**Review by vidai-msft — CHANGES_REQUESTED** [review]

File: src/HdInsightOnAks/HdInsightOnAks.Autorest/README.md (hunk 1 above)
Comment (yuc-Li) [reply src/HdInsightOnAks/HdInsightOnAks.Autorest/README.md:56]: It's just tested locally, and it's been commented out. There is commit id on it to get the file from swagger repo.
image

Comment (vidai-msft) [reply src/HdInsightOnAks/HdInsightOnAks.Autorest/README.md:56]: Please remove the test location from the file. Thanks!

Comment (yuc-Li) [reply src/HdInsightOnAks/HdInsightOnAks.Autorest/README.md]: Removed, thanks.

File: src/HdInsightOnAks/HdInsightOnAks.Autorest/README.md [hunk 2]
```diff
@@ -49,14 +46,12 @@ In this directory, run AutoRest:
 ### General settings
 > Values
 ``` yaml
-commit: f09aacf4c6b63be416212cb182f6b31e8bc6d545
+tag: package-preview-2023-11
```
Comment (vidai-msft) [inline src/HdInsightOnAks/HdInsightOnAks.Autorest/README.md:49]: What is the use of the tag here?

Comment (yuc-Li) [reply src/HdInsightOnAks/HdInsightOnAks.Autorest/README.md:49]: Our [Swagger](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/hdinsight/resource-manager/Microsoft.HDInsight/HDInsightOnAks/readme.md) has been updated to version package-2024-05. In this release, we use package-preview-2023-11.

File: src/HdInsightOnAks/HdInsightOnAks.Autorest/custom/New-AzHdInsightOnAksClusterPoolAKSUpgradeObject.ps1 [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+Create an object to hold the cluster pool upgrade parameters.
+
+.Outputs
+Microsoft.Azure.PowerShell.Cmdlets.HdInsightOnAks.Models.ClusterPoolUpgrade
+.Link
+https://learn.microsoft.com/powershell/module/az.hdinsightonaks/New-AzHdInsightOnAksClusterPoolAKSUpgradeObject
+#>
+function New-AzHdInsightOnAksClusterPoolAKSUpgradeObject {
+    [OutputType([Microsoft.Azure.PowerShell.Cmdlets.HdInsightOnAks.Models.IClusterPoolUpgrade])]
+    [CmdletBinding(DefaultParameterSetName = 'Create', PositionalBinding = $false, SupportsShouldProcess)]
```
Comment (vidai-msft) [inline src/HdInsightOnAks/HdInsightOnAks.Autorest/custom/New-AzHdInsightOnAksClusterPoolAKSUpgradeObject.ps1:28]: This looks an in-memory cmdlet. `SupportsShouldProcess` is not necessary as no server side changes will be made.

File: src/HdInsightOnAks/HdInsightOnAks.Autorest/custom/New-AzHdInsightOnAksClusterHotfixUpgradeObject.ps1 [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+Create an object to hold the cluster upgrade parameters.
+
+.Outputs
+Microsoft.Azure.PowerShell.Cmdlets.HdInsightOnAks.Models.ClusterUpgrade
+.Link
+https://learn.microsoft.com/powershell/module/az.hdinsightonaks/New-AzHdInsightOnAksClusterHotfixUpgradeObject
+#>
+function New-AzHdInsightOnAksClusterHotfixUpgradeObject {
+    [OutputType([Microsoft.Azure.PowerShell.Cmdlets.HdInsightOnAks.Models.IClusterUpgrade])]
+    [CmdletBinding(DefaultParameterSetName = 'Create', PositionalBinding = $false, SupportsShouldProcess)]
```
Comment (vidai-msft) [inline src/HdInsightOnAks/HdInsightOnAks.Autorest/custom/New-AzHdInsightOnAksClusterHotfixUpgradeObject.ps1:28]: This looks an in-memory cmdlet. `SupportsShouldProcess` is not necessary as no server side changes will be made.

File: src/HdInsightOnAks/HdInsightOnAks.Autorest/custom/New-AzHdInsightOnAksClusterPoolAKSUpgradeObject.ps1 (hunk 3 above)
Comment (yuc-Li) [reply src/HdInsightOnAks/HdInsightOnAks.Autorest/custom/New-AzHdInsightOnAksClusterPoolAKSUpgradeObject.ps1]: Removed unnecessary `SupportsShouldProcess `.

File: src/HdInsightOnAks/HdInsightOnAks.Autorest/custom/New-AzHdInsightOnAksClusterHotfixUpgradeObject.ps1 (hunk 4 above)
Comment (yuc-Li) [reply src/HdInsightOnAks/HdInsightOnAks.Autorest/custom/New-AzHdInsightOnAksClusterHotfixUpgradeObject.ps1]: Removed unnecessary `SupportsShouldProcess `.

**Review by vidai-msft — APPROVED** [review]
LGTM

**Pull request merged**